crypto/src/utils.rs

```diff
@@ -0,0 +1,205 @@
+        let mut rng = thread_rng();
+        let mut eraser = Eraser::new();
+        let seed = &secret_key[..SEED_LENGTH];
+        eraser.add_suite::<Sodium>();
+        let pwbox = Sodium::build_box(&mut rng)
+            .seal(pass_phrase, &seed)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't create a pw box"))?;
+        let encrypted_key = eraser
+            .erase(&pwbox)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't convert a pw box"))?;
+
+        Ok(EncryptedKeys {
+            public_key,
+            secret_key: encrypted_key,
+        })
+    }
+
+    fn decrypt(self, pass_phrase: &[u8]) -> Result<(PublicKey, SecretKey), Error> {
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let restored = match eraser.restore(&self.secret_key) {
+            Ok(restored) => restored,
+            Err(_) => {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    "Couldn't restore a secret key",
+                ))
+            }
+        };
+        assert_eq!(restored.len(), SEED_LENGTH);
+        let seed_bytes = restored
+            .open(pass_phrase)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't open an encrypted key"))?
+            .to_vec();
+        let seed = Seed::from_slice(&seed_bytes[..])
+            .ok_or_else(|| Error::new(ErrorKind::Other, "Couldn't create seed from slice"))?;
+        let (public_key, secret_key) = gen_keypair_from_seed(&seed);
+        assert_eq!(self.public_key, public_key);
+        Ok((public_key, secret_key))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    extern crate tempdir;
+
+    use self::tempdir::TempDir;
+    use super::*;
+    use toml;
+
+    #[test]
+    fn test_create_and_read_keys_file() {
+        let dir = TempDir::new("test_utils").expect("Couldn't create TempDir");
+        let file_path = dir.path().join("private_key.toml");
+        let pass_phrase = b"passphrase";
+        let result = create_keys_file(file_path.as_path(), pass_phrase);
+        assert!(result.is_ok());
+        let pk1 = result.unwrap();
+        let result = read_keys_from_file(file_path.as_path(), pass_phrase);
+        assert!(result.is_ok());
+        let (pk2, _) = result.unwrap();
+        assert_eq!(pk1, pk2);
+    }
+
+    #[test]
+    fn test_encrypt_decrypt() {
+        let pass_phrase = b"passphrase";
+        let (pk, sk) = gen_keypair();
+
+        let encrypt_key =
+            EncryptedKeys::encrypt(pk, &sk, pass_phrase).expect("Couldn't encrypt keys");
+
+        let (_, decrypted_key) = encrypt_key
+            .decrypt(pass_phrase)
+            .expect("Couldn't decrypt key");
+
+        assert_eq!(sk, decrypted_key);
+    }
+
+    #[test]
+    fn test_decrypt_from_file() {
+        let pass_phrase = b"passphrase";
+        let file_content = r#"
+public_key = '2e9d0b7ff996acdda58dd786950dec7361d3d81fd188cb250fd0cab2d064aaf8'
```

Comment:
Minor nit: Is there particular reason to de-indent `file_content`? It seems to be working just fine with indentation.

Reply:
Reminder that indentation doesn't seem to hurt readability of TOML (see [this test](https://github.com/exonum/pwbox-rs/blob/b578303f7a8d02bd5b3f11874f35f48b95c4aef8/tests/serialization.rs#L134) for an example).